src/main/java/org/scijava/ui/DefaultUIService.java

```diff
@@ -212,9 +213,9 @@ public boolean isHeadless() {
 
 	@Override
 	public UserInterface getDefaultUI() {
+		if (isHeadless()) return null;
```

Comment:
I really like this. It makes total sense to return `null` in a headless context. And if downstream code can't deal with it—tough. 😝